frontend/src/components/login-form/login.form.jsx

```diff
@@ -7,17 +7,45 @@ import AuthService from "../../pages/api/authService";
 
 export const LoginForm = () => {
   const router = useRouter();
+
+  const [emailValid, setEmailValid] = useState(true);
+  const [passwordValid, setPasswordValid] = useState(true);
+
   const [formData, setFormData] = useState({
     email: "",
     password: "",
   });
 
+  const validateEmail = () => {
+    let result = RegExp("^(.+)@(.+)$").test(formData.email);
+
+    console.log("email", result);
+    if (result) setEmailValid(true);
+    else setEmailValid(false);
+
+    return result;
+  };
+
+  const validatePassword = () => {
+    let result = formData.password != "" ? true : false;
+    console.log("pass", result);
+
+    if (result) setPasswordValid(true);
+    else setPasswordValid(false);
+
+    return result;
+  };
+
   async function submit() {
     try {
-      await AuthService.login(formData.email, formData.password);
-      await router.push("/");
+      if (validateEmail() & validatePassword()) {

```

Comment:
Nema potrebe da bilo šta validiraš pre nego što dobiješ response sa BE. 
Tj. nemoj se uopšte opterećivati sa validacijom na FE. 
Dovoljno je samo da ispišeš "E-mail adresa i/ili lozinka nisu ispravni" poruku ukoliko BE vrati response 400.
Takodje, obiji oba polja sa crvenom bojom u tom slucaju (pogledaj kako izgledaju polja sa errorima u formi za registraciju).

Iako `emailValid` i `passwordValid` nisu potrebni, ubuduće, gledaj da stavljaš odgovaraju prefix za boolean varijable (`is`, `has`, `should` itd.). 
Bolji naziv za ove dve varijable bi bilo: `hasValidEmail` i `hasValidPassword`

Reply:
@milos1302 Vazi, validaciju na FE sam ubacio da se ne bi slao api request u slucaju da email nije validnog formata ili da je lozinka prazna. Cisto radi optimizacije. Mogu ukloniti ako mislis da treba, ali ja bih ostavio.